### PR TITLE
pdpb: Add err type and input param to BatchScanRegions

### PR DESCRIPTION
**Pull request by okJiang**
ref https://github.com/tikv/pd/issues/8358

### CODE REVIEW

File: pkg/raft_serverpb/raft_serverpb.pb.go
Comment (okJiang) [inline pkg/raft_serverpb/raft_serverpb.pb.go:1]: Automatically generated

File: proto/pdpb.proto [hunk 1]
```diff
@@ -377,6 +378,9 @@ message BatchScanRegionsRequest {
 
     repeated KeyRange ranges = 3; // the given ranges must be in order.
     int32 limit = 4; // limit the total number of regions to scan.
+    // If output_must_contain_all_key_range is true, the output must contain all key ranges in the request.
+    // If the output does not contain all key ranges, the request is considered failed and returns an error(REGIONS_NOT_CONTAIN_ALL_KEY_RANGE).
+    bool output_must_contain_all_key_range = 5;
```
Comment (rleungx) [inline proto/pdpb.proto:383]: Can we make it shorter?

Comment (okJiang) [reply proto/pdpb.proto:383]: Do you have some better suggestions?😂

File: proto/pdpb.proto [hunk 2]
```diff
@@ -377,6 +378,9 @@ message BatchScanRegionsRequest {
 
     repeated KeyRange ranges = 3; // the given ranges must be in order.
     int32 limit = 4; // limit the total number of regions to scan.
+    // If output_must_contain_all_key_range is true, the output must contain all key ranges in the request.
+    // If the output does not contain all key ranges, the request is considered failed and returns an error(REGIONS_NOT_CONTAIN_ALL_KEY_RANGE).
```
Comment (overvenus) [inline proto/pdpb.proto:382]:
```suggestion
    // If output_must_contain_all_key_range is true, the output must contain all
    // key ranges in the request.
    // If the output does not contain all key ranges, the request is considered
    // failed and returns an error(REGIONS_NOT_CONTAIN_ALL_KEY_RANGE).
```

Prefer 80 character columns.

File: proto/pdpb.proto (hunk 1 above)
Comment (rleungx) [reply proto/pdpb.proto:385]: contain_all_key_range?

**Review by rleungx — APPROVED** [review]

File: proto/pdpb.proto (hunk 1 above)
Comment (okJiang) [reply proto/pdpb.proto:385]: What do you think which is better? @lance6716 @you06

Comment (you06) [reply proto/pdpb.proto:385]: With the comment explaination, `contain_all_key_range` is clear enough to me.

Comment (okJiang) [reply proto/pdpb.proto]: @you06 updated in https://github.com/pingcap/kvproto/pull/1252/commits/49bf52b9e60090df9f0eb3040a2ce7ae84edbe4f

**Comment by okJiang** [comment]
如果没有其他问题的话，麻烦给我一个 approve

If you have no question, could you please give me an approve? @you06

**Review by you06 — APPROVED** [review]

**Comment by ekexium** [comment]
We'd better merge such changes together with the related TiKV PRs to prevent breaking TiKV's build when using the master kvproto.

**Comment by okJiang** [comment]
> We'd better merge such changes together with the related TiKV PRs to prevent breaking TiKV's build when using the master kvproto.

got it

**Pull request merged**